src/components/report-technical-issue/template.njk

```diff
@@ -4,5 +4,5 @@
   class="govuk-link hmrc-report-technical-issue {{ params.classes }}"
   {% if params.referrerUrl %} rel="noreferrer noopener"{% endif %}
   target="_blank"
-  href="{{ params.baseUrl }}/contact/report-technical-problem?newTab=true{% if serviceId %}&amp;service={{ serviceId | urlencode }}{% endif %}{% if params.referrerUrl %}&amp;referrerUrl={{ params.referrerUrl | urlencode }}{% endif %}"
+  href="{{ params.baseUrl }}/contact/report-technical-problem{% if serviceId or params.referrerUrl %}?{% endif %}{% if serviceId %}service={{ serviceId | urlencode }}{% endif %}{% if params.referrerUrl %}&amp;referrerUrl={{ params.referrerUrl | urlencode }}{% endif %}"
```

Comment:
ah, I think we might need to tweak this

if you have no service id but you do have a referrer url then you'll get a url like `?&amp;referrerUrl` which might be ok - I'm not sure - but should be `?referrerUrl` in that case

Reply:
logic is something like

```
if serviceId then "?serviceId=$serviceId"

if referrerUrl then
  (if serviceId then "&amp;" else "?")
    + "referrerUrl=$referrerUrl"
```

but might be nicer ways of writing it 😅

Reply:
```nunjucks 
  {% set urlBuilder %}
  {% if serviceId %}?service={{ serviceId | urlencode }}{% endif %}
  {% if params.referrerUrl and serviceId %}&amp;{% endif %}
  {% if params.referrerUrl and not serviceId %}?{% endif %}
  {% if params.referrerUrl %}referrerUrl={{ params.referrerUrl | urlencode }}{% endif %}
  {% endset %}
```
```nunjucks
  href="{{ params.baseUrl }}/contact/report-technical-problem{{urlBuilder}}"
```
This seems to work with a few tweaks to the tests to accommodate. Separating it out from the href param feels less confusing. Thoughts?

Reply:
Will update `urlBuilder` to `urlParametersBuilder`